avocado/core/output.py

```diff
@@ -557,7 +557,7 @@ def __del__(self):
     def close(self):
         try:
             self.pipe.close()
-        except OSError:
+        except (OSError, AttributeError):
```

Comment:
How can `self.pipe` not exist, if it's set unconditionally at `__init__()` time?

Reply:
@clebergnu yes, it was strange to me too, but I was getting this error when trying to run `make check-full`.

Reply:
@clebergnu https://paste.centos.org/view/d18d805c

Reply:
So I was wrong and `self.pipe` is not unconditionally set because when the `RuntimeError` exception is raised `selfpipe` is never set.  Then, at garbage collection time, `self.close()` and tries to act on the `self.pipe` anyway.  IMO this is a more honest approach:

```diff
diff --git a/avocado/core/output.py b/avocado/core/output.py
index 4a871206..e3b5d644 100644
--- a/avocado/core/output.py
+++ b/avocado/core/output.py
@@ -542,6 +542,7 @@ class Paginator:
     """
 
     def __init__(self):
+        self.pipe = None
         paginator = os.environ.get('PAGER')
         if not paginator:
             try:
@@ -555,20 +556,23 @@ class Paginator:
         self.close()
 
     def close(self):
-        try:
-            self.pipe.close()
-        except OSError:
-            pass
+        if self.pipe is not None:
+            try:
+                self.pipe.close()
+            except OSError:
+                pass
 
     def write(self, msg):
-        try:
-            self.pipe.write(msg)
-        except (OSError, ValueError):
-            pass
+        if self.pipe is not None:
+            try:
+                self.pipe.write(msg)
+            except (OSError, ValueError):
+                pass
 
     def flush(self):
-        if not self.pipe.closed:
-            self.pipe.flush()
+        if self.pipe is not None:
+            if not self.pipe.closed:
+                self.pipe.flush()
 
 
 def add_log_handler(logger, klass=logging.StreamHandler, stream=sys.stdout,
```

What do you think?

Reply:
Yes, I think that is valid, but I would suggest removing the "is not None". If self.pipe: ....

Reply:
Also looks good to me.  Will you send a v2?

Reply:
Yes, just doing.